libraries/aws_iam_saml_provider.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsIamSamlProvider < AwsResourceBase
+  name 'aws_iam_saml_provider'
+  desc 'Verifies settings for a SAML Provider.'
+  example '
+    describe aws_iam_saml_provider("arn:aws:iam::123456789012:saml-provider/FANCY") do
+        it { should exist }
+        its("arn") { should match("arn:aws:iam::.*:saml-provider\/FANCY") }
+        its("valid_until") { should be > Time.now + 90 * 86400 }
+    end
+  '
+  supports platform: 'aws'
+
+  attr_reader :provider, :arn, :create_date, :valid_until
+
+  def initialize(opts = {})
+    opts = { saml_provider_arn: opts } if opts.is_a?(String)
+    super(opts)
+    validate_parameters([:saml_provider_arn])
+
+    catch_aws_errors do
+      @provider = @aws.iam_client.get_saml_provider(saml_provider_arn: opts[:saml_provider_arn])
+      @arn = opts[:saml_provider_arn]
+      @create_date = @provider.create_date
+      @valid_until = @provider.valid_until
+      # Not currently doing anything with metadata doc, future improvement?
+      # @saml_metadata_document = @provider.saml_metadata_document
```

Comment:
I would suggest simply exposing `saml_metadata_document` now - save us revisiting in the future. Once it's exposed we are then exposing everything returned from the API and can therefor make `@provider` a local variable and remove it as an `attr_reader`.